### PR TITLE
Remove tls-sni challenge in manual plugin

### PR DESCRIPTION
**Pull request by adferrand**
This PR is a part of the tls-sni-01 removal plan described in #6849.

It removes the capability to make tls-sni-01 challenges in the manual plugin.

### CODE REVIEW

File: certbot/plugins/manual.py [hunk 1]
```diff
@@ -191,16 +143,9 @@ def _perform_achall_with_script(self, achall):
...
-            env['CERTBOT_SNI_DOMAIN'] = self.tls_sni_01.get_z_domain(achall)
-            os.environ.pop('CERTBOT_VALIDATION', None)
-            env.pop('CERTBOT_VALIDATION')
-        else:
-            os.environ.pop('CERTBOT_CERT_PATH', None)
-            os.environ.pop('CERTBOT_KEY_PATH', None)
-            os.environ.pop('CERTBOT_SNI_DOMAIN', None)
+        os.environ.pop('CERTBOT_CERT_PATH', None)
+        os.environ.pop('CERTBOT_KEY_PATH', None)
+        os.environ.pop('CERTBOT_SNI_DOMAIN', None)
```
Comment (bmw) [inline certbot/plugins/manual.py]: I think we can remove unsetting these environment variables as we only set them in the now deleted code for TLS-SNI-01.

File: certbot/plugins/manual.py [hunk 2]
```diff
@@ -213,17 +158,10 @@ def _perform_achall_manually(self, achall):
                 achall=achall, encoded_token=achall.chall.encode('token'),
                 port=self.config.http01_port,
                 uri=achall.chall.uri(achall.domain), validation=validation)
-        elif isinstance(achall.chall, challenges.DNS01):
+        else:  # Can be only DNS-01
             msg = self._DNS_INSTRUCTIONS.format(
```
Comment (bmw) [inline certbot/plugins/manual.py]: nit: Want to add `assert isinstance(achall.chall, challenges.DNS01)` above this line like we had for `TLSSNI01`? I think this accomplishes the same thing as your comment and will cause us to cleanly error out if something has gone horribly wrong.

File: certbot/plugins/manual_test.py [hunk 3]
```diff
@@ -81,16 +77,14 @@ def test_script_perform(self):
             'print(os.environ.get(\'CERTBOT_TOKEN\', \'notoken\'));'
             'print(os.environ.get(\'CERTBOT_CERT_PATH\', \'nocert\'));'
             'print(os.environ.get(\'CERTBOT_KEY_PATH\', \'nokey\'));'
```
Comment (bmw) [inline certbot/plugins/manual_test.py]: Can we remove the lines for `CERTBOT_CERT_PATH` and `CERTBOT_KEY_PATH`?

File: certbot/plugins/manual.py (hunk 1 above)
Comment (adferrand) [reply certbot/plugins/manual.py]: Done.

File: certbot/plugins/manual.py (hunk 2 above)
Comment (adferrand) [reply certbot/plugins/manual.py]: Done.

File: certbot/plugins/manual_test.py (hunk 3 above)
Comment (adferrand) [reply certbot/plugins/manual_test.py]: Done.

**Pull request merged**